manifests/params.pp

```diff
@@ -31,6 +31,12 @@
     $service_enable = true
   }
 
+  if ($::operatingsystem == 'Ubuntu' and $::operatingsystemrelease in ['15.04','15.10','16.04','16.10']) or ($::operatingsystem == 'Debian' and $::operatingsystemrelease == '8') {
```

Comment:
Using `backticks/%x` or `Kernel.exec` will ignore the confine, so it will run on any system rather than be restricted to Linux (eg. Windows, OSX, BSD).

Good spot, It should be `Facter::Util::Resolution.exec` which is backwards compatible:
http://www.rubydoc.info/gems/facter/1.7.5/Facter/Util/Resolution.exec
http://www.rubydoc.info/gems/facter/2.4.6/Facter/Util/Resolution.exec

```ruby
Facter.add(:gitlab_systemd) do
  confine :kernel => :linux
  setcode do
     Facter::Util::Resolution.exec('ps -p 1 -o comm=') == 'systemd'
  end
```
end

I'll open a PR against the systemd repo for that to fix that! 👍